config/src/telemetry.rs

```diff
@@ -17,36 +16,26 @@ pub struct Configuration {
     #[config(serde_as_str)]
     pub url: Option<Url>,
     /// The minimum period of time in seconds to wait before reconnecting
-    #[serde(default = "default_min_retry_period")]
     pub min_retry_period: u64,
     /// The maximum exponent of 2 that is used for increasing delay between reconnections
-    #[serde(default = "default_max_retry_delay_exponent")]
     pub max_retry_delay_exponent: u8,
     /// The filepath that to write dev-telemetry to
     #[config(serde_as_str)]
     pub file: Option<std::path::PathBuf>,
 }
 
-impl Default for Configuration {
+impl Default for ConfigurationProxy {
```

Comment:
Since we don't actually use this outside the compile-time context, we might consider using the `ConstDefault` trait:

```rust
pub trait ConstDefault {
     const DEFAULT: Self;
}

Reply:
To be dealt in #2746.